sympy/solvers/polysys.py

```diff
@@ -228,7 +247,16 @@ def _subs_root(f, gen, zero):
     def _solve_reduced_system(system, gens, entry=False):
         """Recursively solves reduced polynomial systems. """
         if len(system) == len(gens) == 1:
-            zeros = list(roots(system[0], gens[-1]).keys())
+            rts = roots(system[0], gens[-1])
+
+            if strict and sum(rts.values()) < degree(system[0], gens[-1]):
+                raise NotImplementedError(filldedent('''
+                only systems for which every polynomial in the Groebner basis
+                has all its solutions expressible in radicals (after it is converted
+                to a univariate polynomial) are supported
```

Comment:
This should be a sentence with capital letter and full stop. Also I'd change this to say something like:

Strict mode: some solution components cannot be expressed in radicals so a complete set of solutions cannot be returned. Call solve_poly_system with strict=False to see partial solutions (if there are any).